theano/gpuarray/type.py

```diff
@@ -572,6 +592,8 @@ def gpuarray_shared_constructor(value, name=None, strict=False,
     See :func:`theano.shared`.
 
     """
+    if target is None and not move_to_gpu(value):
```

Comment:
Perhaps move this check until after we know it's an array.  Otherwise it can crash move_to_gpu since the value might not have a dtype or ndim attribute.